includes/class-wc-amazon-payments-advanced-rest-api-controller.php

```diff
@@ -549,15 +628,19 @@ protected function get_missing_capture_id_request_error( $request ) {
 	/**
 	 * Check whether order is valid to proceed.
 	 *
-	 * @param WP_Post $order_post Order post object.
+	 * @param int $order_post Order post object.
 	 *
-	 * @return bool True if it's valid request.
+	 * @return WP_Post|WP_Error Post object if it's valid, WP_Error if it's invalid.
 	 */
-	protected function is_valid_order( $order_post ) {
+	protected function is_valid_order( $order_id ) {
+		$order_post = get_post( (int) $order_id );
```

Comment:
why not use the WC_Order object to do the verifications in this method?

Reply:
Keeping the same implementation as v1 for backwards compatibility